doc/api/esm.md

```diff
@@ -196,664 +196,24 @@ unspecified.
-    import Date from 'date';
-    const someDate = new Date();
-    // someDate contains state; Date does not
-    ```
-
-   The `new` keyword isn’t required; a package’s function can return a new
-   object, or modify a passed-in object, to keep the state external to the
-   package.
-
-1. Isolate the state in one or more CommonJS files that are shared between the
-   CommonJS and ES module versions of the package. For example, if the CommonJS
-   and ES module entry points are `index.cjs` and `index.mjs`, respectively:
-
-    ```js
-    // ./node_modules/pkg/index.cjs
-    const state = require('./state.cjs');
-    module.exports.state = state;
-    ```
-
-    ```js
-    // ./node_modules/pkg/index.mjs
-    import state from './state.cjs';
-    export {
-      state
-    };
-    ```
-
-   Even if `pkg` is used via both `require` and `import` in an application (for
-   example, via `import` in application code and via `require` by a dependency)
-   each reference of `pkg` will contain the same state; and modifying that
-   state from either module system will apply to both.
-
-Any plugins that attach to the package’s singleton would need to separately
-attach to both the CommonJS and ES module singletons.
-
-This approach is appropriate for any of the following use cases:
-* The package is currently written in ES module syntax and the package author
-  wants that version to be used wherever such syntax is supported.
-* The package is stateless or its state can be isolated without too much
-  difficulty.
-* The package is unlikely to have other public packages that depend on it, or if
-  it does, the package is stateless or has state that need not be shared between
-  dependencies or with the overall application.
-
-Even with isolated state, there is still the cost of possible extra code
-execution between the CommonJS and ES module versions of a package.
-
-As with the previous approach, a variant of this approach not requiring
-conditional exports for consumers could be to add an export, e.g.
-`"./module"`, to point to an all-ES module-syntax version of the package:
-
-<!-- eslint-skip -->
-```js
-// ./node_modules/pkg/package.json
-{
-  "type": "module",
-  "main": "./index.cjs",
-  "exports": {
-    ".": "./index.cjs",
-    "./module": "./index.mjs"
-  }
-}
-```
+<!-- Anchors to make sure old links find a target -->
+<a id="esm_package_entry_points"></a>
+<a id="esm_main_entry_point_export"></a>
+<a id="esm_subpath_exports"></a>
+<a id="esm_package_exports_fallbacks"></a>
+<a id="esm_exports_sugar"></a>
+<a id="esm_conditional_exports"></a>
+<a id="esm_nested_conditions"></a>
+<a id="esm_self_referencing_a_package_using_its_name"></a>
+<a id="esm_internal_package_imports"></a>
+<a id="esm_dual_commonjs_es_module_packages"></a>
+<a id="esm_dual_package_hazard"></a>
+<a id="esm_writing_dual_packages_while_avoiding_or_minimizing_hazards"></a>
+<a id="esm_approach_1_use_an_es_module_wrapper"></a>
+<a id="esm_approach_2_isolate_state"></a>
```

Comment:
Have you already considered making these into individual clickable links to each section? That would be more like what I had in mind. If this is already causing an inconvenience, I think we should at least make it as seemless as possible.

Reply:
Please reconsider this, I really think that would set a bad precedent if we forbid ourselves to ever move an experimental documentation.
With the current version, links to those sections that are not set to an exact version would be a bit still supported, while it limits the visual bloat for users of future generations who won't care that the documentation used to be ordered differently once upon a time.

Nonetheless, if you feel strongly about this, I'm open to work on it.

Reply:
> Nonetheless, if you feel strongly about this, I'm open to work on it.

I don't really feel strongly about this, but it seems like you may have already considered what I had in mind. I think it would be best to wait and see what others' opinions are on the matter. I think either way would be fine by me.